docs/HTTP_API.md

```diff
@@ -1705,6 +1791,91 @@ The handler-specific descriptions below describe how each handler populates the
     {"meta":{"status":"OK","type":"application/json"},"data":{result:[{"name":"Name","description":"Recording name","defaultValue":"Recording"},{"name":"Duration","description":"Duration of recording","defaultValue":"30s[s]"},{"name":"Max Size","description":"Maximum size of recording","defaultValue":"0B[B]"},{"name":"Max Age","description":"Maximum age of the events in the recording","defaultValue":"0s[s]"},{"name":"To disk","description":"Record to disk","defaultValue":"false"},{"name":"Dump on Exit","description":"Dump recording data to disk on JVM exit","defaultValue":"false"}]}}
     ```
 
+* #### `TargetRecordingGetHandler`
+
+    ###### synopsis
+    Returns a recording of a target JVM as an octet stream, using a JWT for
+    authorization. See `AuthTokenPostHandler`.
+
+    ###### request
+    `GET /api/v2.1/targets/:targetId/recordings/:recordingName?token=:jwt`
+
+    `targetId` - The location of the target JVM to connect to,
+    in the form of a `service:rmi:jmx://` JMX Service URL, or `hostname:port`.
+    Should use percent-encoding.
+
+    `recordingName` - The name of the recording to get.
+    Should use percent-encoding.
+
+    `jwt` - The JSON Web Token providing authorization for this request. See
+    `AuthTokenPostHandler`.
+
+    ###### response
+    `200` - The body is an octet stream consisting of the requested recording.
+
+    `401` - User authentication failed. The body is an error message.
+
+    `404` - The target or the recording could not be found.
+    The body is an error message.
+
+    `427` - JMX authentication failed. The body is an error message.
+    There will be an `X-JMX-Authenticate: $SCHEME` header that indicates
+    the authentication scheme that is used.
+
+    `502` - JMX connection failed. This is generally because the target
+    application has SSL enabled over JMX, but Cryostat does not trust the
+    certificate.
+
+    ###### example
+    ```
+    $ curl localhost:8181/api/v2.1/targets/localhost/recordings/foo?token=(trimmed) --output foo.jfr
+      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                     Dload  Upload   Total   Spent    Left  Speed
+    100  530k    0  530k    0     0  9303k      0 --:--:-- --:--:-- --:--:-- 9303k
+    ```
+
+* #### `TargetReportGetHandler`
+
+    ###### synopsis
+    Returns a report of a recording of a target JVM.
+
+    ###### request
+    `GET /api/v2.1/targets/:targetId/reports/:recordingName?token=:jwt`
+
+    `targetId` - The location of the target JVM to connect to,
+    in the form of a `service:rmi:jmx://` JMX Service URL, or `hostname:port`.
+    Should use percent-encoding.
+
+    `recordingName` - The name of the recording to get the report for.
+    Should use percent-encoding.
+
+    `jwt` - The JSON Web Token providing authorization for this request. See
+    `AuthTokenPostHandler`.
+
+    ###### response
+    `200` - The body is the requested report as an HTML document.
```

Comment:
I remember you mentioned about generating in either `.html` or `.json`. Any reference on that feat here?

Reply:
https://github.com/cryostatio/cryostat/pull/1009

Reply:
The reports sidecar is able to do that already, but this endpoint on the main Cryostat API gateway doesn't support it (yet). Needs #1009 which needs #1072 first.

Reply:
Right! Docs can be updated after those then!